src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +125,24 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
-
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
-
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
-
-      context.parent ! Started(ActorSubscriber(stdin), ActorPublisher(stdout), ActorPublisher(stderr))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => self ! StdinTerminated)
+          futureResult
+        }
+
+      val stdout = Source.inputStream(process.getInputStream).withAttributes(Attributes.inputBuffer(1, 1))
+        .mapMaterializedValue { futureResult =>
```

Comment:
A little terser to do: `.mapMaterializedValue { _ andThen { case _ => self ! StdoutTerminated } }`